conference_scheduler/lp_problem/constraints.py

```diff
@@ -41,15 +42,16 @@ def _events_in_session_share_a_tag(events, slots, X):
     for session in session_indices:
         slots = lpu._slots_in_session(session, session_array)
         for slot, event in it.product(slots, event_indices):
-            other_events = lpu._events_with_diff_tag(event, tag_array)
-            for other_slot, other_event in it.product(slots, other_events):
-                if other_slot != slot and other_event != event:
-                    # If they have different tags they cannot be scheduled
-                    # together
-                    yield Constraint(
-                        f'{label} - event: {event}, slot: {slot}',
-                        X[(event, slot)] + X[(other_event, other_slot)] <= 1
-                    )
+            if events[event].tags is not []:
```

Comment:
Did you mean to test identity here, or just equality? Since the second half of the comparison will be created every time, I think this always returns True:

```pycon
>>> [] is not []
True
>>> x = []
>>> x is not []
True
```